moveit_ros/moveit_servo/tests/test_utils.cpp

```diff
@@ -0,0 +1,255 @@
+{
+  Eigen::VectorXd invalid_vector(6);
+  invalid_vector << 0.0, 0.0, 0.0, 0.0, std::nan("");
+  moveit_servo::Twist invalid_twist;
+  invalid_twist.velocities = invalid_vector;
+  EXPECT_FALSE(moveit_servo::isValidCommand(invalid_twist));
+}
+
+TEST(ServoUtilsUnitTests, invalidTwistVelocities)
+{
+  Eigen::VectorXd invalid_vector(6);
+  invalid_vector << 0.0, 0.0, 0.0, 0.0, 0.0, std::nan("");
+  moveit_servo::Twist invalid_twist{ "panda_link0", invalid_vector };
+  EXPECT_FALSE(moveit_servo::isValidCommand(invalid_twist));
+}
+
+TEST(ServoUtilsUnitTests, validIsometry)
+{
+  Eigen::Isometry3d valid_isometry;
+  valid_isometry.setIdentity();
+  EXPECT_TRUE(moveit_servo::isValidCommand(valid_isometry));
+}
+
+TEST(ServoUtilsUnitTests, invalidIsometry)
+{
+  Eigen::Isometry3d invalid_isometry;
+  invalid_isometry.setIdentity();
+  invalid_isometry.translation().z() = std::nan("");
+  EXPECT_FALSE(moveit_servo::isValidCommand(invalid_isometry));
+}
+
+TEST(ServoUtilsUnitTests, validPose)
+{
+  Eigen::Isometry3d valid_isometry;
+  valid_isometry.setIdentity();
+  moveit_servo::Pose valid_pose{ "panda_link0", valid_isometry };
+  EXPECT_TRUE(moveit_servo::isValidCommand(valid_pose));
+}
+
+TEST_F(ServoCppFixture, testVeclocityScaling)
+{
+  moveit::core::JointBoundsVector joint_bounds = joint_model_group_->getActiveJointModelsBounds();
+  // Get the upper bound for the velocities of each joint.
+  Eigen::VectorXd incoming_velocities(joint_bounds.size());
+  for (size_t i = 0; i < joint_bounds.size(); i++)
+  {
+    const auto joint_bound = (*joint_bounds[i])[0];
+    if (joint_bound.velocity_bounded_)
+    {
+      incoming_velocities(i) = joint_bound.max_velocity_;
+    }
+  }
+
+  // Create incoming velocities with only joint 1 and joint 7 over limit by a factor of 0.1 and 0.05
+  // Scale all other joint velocities by 0.3 to keep it within limits.
+  incoming_velocities(0) += (incoming_velocities(0) * 0.1);
+  incoming_velocities(1) -= (incoming_velocities(1) * 0.3);
+  incoming_velocities(2) -= (incoming_velocities(2) * 0.3);
+  incoming_velocities(3) -= (incoming_velocities(3) * 0.3);
+  incoming_velocities(4) -= (incoming_velocities(4) * 0.3);
+  incoming_velocities(5) -= (incoming_velocities(5) * 0.3);
+  incoming_velocities(6) += (incoming_velocities(6) * 0.05);
+
+  // The resulting scaling factor selected should be 0.9, but will be 0.90909094 due to float precision.
+  double user_velocity_override = servo_param_listener_->get_params().override_velocity_scaling_factor;
+  double scaling_factor =
+      moveit_servo::velocityScalingFactor(incoming_velocities, joint_bounds, user_velocity_override);
+  ASSERT_FLOAT_EQ(scaling_factor, 0.90909094);
+}
+
+TEST_F(ServoCppFixture, testLeavingSingularity)
+{
+  Eigen::VectorXd cartesian_delta(6);
+  cartesian_delta << 0.005, 0.0, 0.0, 0.0, 0.0, 0.0;
+
+  // Home state
+  Eigen::VectorXd home_state(7);
+  home_state << 0.0, -0.785, 0.0, -2.356, 0.0, 1.571, 0.785;
```

Comment:
Can you get this directly from a named state from the `robot_state_` variable? No big deal either way.

Reply:
YES ! , the configuration is available as the named state "ready". 
Thanks for pointing it out.